File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-3.js
class Solution {
    /**
     * @param {string} s
     * @param {number} k
     * @return {number}
     */
    characterReplacement(s, k) {
        let l = 0;
        let count = {};
        let longest = 0;
        for (let r = 0; r < s.length; r ++) {
            // every iteration will increase one count
            count[s[r]] = (count[s[r]] || 0) + 1;
            let distance = r - l + 1;
            if ( distance - Math.max(...Object.values(count)) > k) {
                count[s[l]] = (count[s[l]] - 1)
                l ++
            }
            // calculate distance between two pointers
            distance = r - l + 1;
            longest = Math.max(longest, distance)
        }

        return longest
    }
}

File: Data Structures & Algorithms/car-pooling/submission-3.py
class Solution:
    def carPooling(self, trips: List[List[int]], capacity: int) -> bool:
        end = 0 

        for _, _, to in trips: 
            end = max(end, to) 

        road = [ 0 ] * ( end + 1 ) 

        for n, u, v in trips: 
            road[u] += n 
            road[v] -= n 

        count = 0 
        for c in road: 
            count += c 
            if count > capacity: 
                return False 

        return True